src/DynamicRupture/FrictionLaws/RateAndState.h

```diff
@@ -10,363 +10,442 @@ namespace seissol::dr::friction_law {
-   * mu = a * arcsinh[ V/(2*V0) * exp(SV/a) ]
-   */
-  void updateMu(unsigned int ltsFace, unsigned int pointIndex, real localStateVariable);
-}; // end class FL_103
-
-/**
- * As a reference, see https://www.scec.org/meetings/2020/am/poster/158
- */
-class RateAndStateThermalPressurizationLaw : public RateAndStateFastVelocityWeakeningLaw {
-  public:
-  using RateAndStateFastVelocityWeakeningLaw::RateAndStateFastVelocityWeakeningLaw;
-
-  protected:
-  real (*temperature)[numPaddedPoints];
-  real (*pressure)[numPaddedPoints];
-  real (*TP_Theta)[numPaddedPoints][TP_grid_nz];
-  real (*TP_sigma)[numPaddedPoints][TP_grid_nz];
-  real (*TP_halfWidthShearZone)[numPaddedPoints];
-  real (*alphaHy)[numPaddedPoints];
-
-  real TP_grid[TP_grid_nz];
-  real TP_DFinv[TP_grid_nz];
-
-  real faultStrength[numPaddedPoints];
-  real Theta_tmp[TP_grid_nz];
-  real Sigma_tmp[TP_grid_nz];
-
-  public:
-  /*
-   * initialize local attributes (used in initializer class respectively)
-   */
-  void initializeTP(seissol::Interoperability& e_interoperability);
-
-  /*
-   * copies all parameters from the DynamicRupture LTS to the local attributes
-   */
-  virtual void copyLtsTreeToLocalRS(seissol::initializers::Layer& layerData,
-                                    seissol::initializers::DynamicRupture* dynRup,
-                                    real fullUpdateTime) override;
-
-  protected:
-  /*
-   * set initial value of thermal pressure
-   */
-  void hookSetInitialP_f(std::array<real, numPaddedPoints>& P_f, unsigned int ltsFace) override;
-
-  /*
-   * compute thermal pressure according to Noda and Lapusta 2010
-   * bool saveTmpInTP is used to save final thermal pressure values for theta and sigma
-   */
   void hookCalcP_f(std::array<real, numPaddedPoints>& P_f,
                    FaultStresses& faultStresses,
                    bool saveTmpInTP,
                    unsigned int timeIndex,
-                   unsigned int ltsFace) override;
-
-  /*
-   * compute thermal pressure according to Noda and Lapusta 2010
-   */
-  void updateTemperatureAndPressure(unsigned int pointIndex,
-                                    unsigned int timeIndex,
-                                    unsigned int ltsFace);
+                   unsigned int ltsFace) {}
 
-  real heatSource(real tmp, real alpha, unsigned int iTP_grid_nz, unsigned int timeIndex);
+  bool IterativelyInvertSR(unsigned int ltsFace,
+                           int nSRupdates,
+                           std::array<real, numPaddedPoints> const& localStateVariable,
+                           std::array<real, numPaddedPoints> const& normalStress,
+                           std::array<real, numPaddedPoints> const& absoluteShearStress,
+                           std::array<real, numPaddedPoints>& slipRateTest) {
+
+    real tmp[numPaddedPoints], tmp2[numPaddedPoints], tmp3[numPaddedPoints], mu_f[numPaddedPoints],
+        dmu_f[numPaddedPoints], NR[numPaddedPoints], dNR[numPaddedPoints];
```

Comment:
I would suggest to split this chain of definitions into 2-3 lines: 3 definitions per line